salt/hosts-minion/init.sls

```diff
@@ -0,0 +1,14 @@
+{%- for minion_id, addrlist in salt['mine.get']('roles:kube-master', 'network.ip_addrs', 'grain').items() %}
+{{ minion_id }}-host-entry:
+  host.present:
+{% if addrlist is string %}
+    - ip: {{ addrlist }}
+{% else %}
+    - ip: {{ addrlist|first }}
+{% endif %}
+    - names:
+      - api
+      - api.{{ pillar['internal_infra_domain'] }}
+      - {{ minion_id }}
+      - {{ minion_id }}.{{ pillar['internal_infra_domain'] }}
+{% endfor %}
```

Comment:
Well, I think that, with this solution, you could DoS your API servers with an scenario like, for example, 7 API servers and 200 workers, and no _"other things in the cluster are going to fall over first"._

Anyway, not a big deal. I will implement this in another PR later on...